spring-cloud-dataflow-composed-task-runner/src/test/java/org/springframework/cloud/dataflow/composedtaskrunner/properties/ComposedTaskPropertiesTests.java

```diff
@@ -121,6 +122,23 @@ public void testComposedTaskAppArguments() {
 				});
 	}
 
+	@Test
+	public void testAssignmentOfOauth2ClientCredentialsClientAuthenticationMethod(){
+		this.contextRunner
+				.withInitializer(context -> {
+					Map<String, Object> map = new HashMap<>();
+					map.put("oauth2ClientCredentialsClientAuthenticationMethod", "POST");
+					context.getEnvironment().getPropertySources().addLast(new SystemEnvironmentPropertySource(
```

Comment:
I am pretty sure you copied the intiializer/properties part from the test above but a couple of points...

* I think we can instead simply use `ApplicationContextRunner.withSystemProperties`

However, if we did continue using the initializer approach:

* I don't think we would need to use `SystemEnvironmentPropertySource` and instead could use a simple `MapPropertySource`
* I am not sure why the `addLast` method is used rather than `addFirst` as we want these props to take precedence (unless the `SystemEnvironmentPropertySource` has a higher built-in order/priority.

Reply:
Hey @onobc - thanks for the review! I used withSystemProperties, because I just want to check if the property is converted into the enum correctly. 👍